docs/source/cpp/streaming_execution.rst

```diff
@@ -647,6 +649,25 @@ SelectK example:
 
 .. _stream_execution_scan_docs:
 
+``table_sink``
+----------------
+
+.. _stream_execution_table_sink_docs:
+
+Considering the variety of sink nodes provided in the streaming execution engine, the ``table_sink`` node 
+provides the ability to take the output as a table. It is much easier to use 
+:class:`arrow::compute::TableSinkNodeOptions`.
+The output data can be obtained as a ``std::shared_ptr<arrow::Table>`` along with the output ``schema``. 
```

Comment:
Technically it was an input schema I think (and it's going away now).  A `Table` already has a schema associated with it so there is no need to specify the output schema separately.